libs/api/ebpf_api.cpp

```diff
@@ -3197,27 +3186,42 @@ ebpf_result_t
 ebpf_get_program_type_by_name(
     _In_z_ const char* name, _Out_ ebpf_program_type_t* program_type, _Out_ ebpf_attach_type_t* expected_attach_type)
 {
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_program_type_t* program_type_uuid;
     EBPF_LOG_ENTRY();
     ebpf_assert(name);
     ebpf_assert(program_type);
     ebpf_assert(expected_attach_type);
 
-    EbpfProgramType type = get_program_type_windows(name, name);
-    ebpf_program_type_t* program_type_uuid = (ebpf_program_type_t*)type.platform_specific_data;
+    try {
+        EbpfProgramType type = get_program_type_windows(name, name);
```

Comment:
fir consistency with line 3220 can this be
```suggestion
        const EbpfProgramType& type = get_program_type_windows(name, name);
```
?

Reply:
fixed.